src/unittest/python/serialization/GeoShapeDeserializer_test.py

```diff
@@ -0,0 +1,57 @@
+# Name: Debasish Kanhar
+
+import unittest
+import json
+from janusgraph_python.structure.io.GraphsonReader import JanusGraphSONReader
+from janusgraph_python.core.datatypes.Point import Point
+from janusgraph_python.core.datatypes.Circle import Circle
+
+
+class TestGeoShapeDeserializer(unittest.TestCase):
+    def setUp(self):
+        self.latitude = 80.0
+        self.longitude = 100.0
+        self.radius = 5
+
+        self.reader = JanusGraphSONReader().build()
+        pass
+
+    def test_point_deserialization(self):
+        pointGSON = "{\"@type\": \"janusgraph:Geoshape\", " \
+                    "\"@value\": {\"coordinates\": [" + str(self.latitude) + ", " + str(self.longitude) + "]}}"
+
+        pointJSON = json.loads(pointGSON)
+
+        expectedPoint = self.reader.toObject(pointJSON)
+
+        actualPoint = Point(self.longitude, self.latitude)
+
+        self.assertEqual(expectedPoint, actualPoint)
+        pass
+
+    def test_circle_deserialization(self):
+
+        circleGSON = "{\"" \
+                     "@type\":\"janusgraph:Geoshape\"," \
+                     "\"@value\":" \
+                        "{\"geometry\":" \
+                            "{\"@type\":\"g:Map\"," \
+                            "\"@value\":" \
+                        "[\"type\",\"Circle\"," \
+                        "\"coordinates\",{" \
+                                            "\"@type\":\"g:List\"," \
+                                            "\"@value\":[" \
+                                                "{\"@type\":\"g:Double\",\"@value\": " + str(self.latitude) + "}," \
+                                                "{\"@type\":\"g:Int32\",\"@value\":" + str(self.longitude) + "}" \
+                                                        "]" \
+                                            "}, " \
+                        "\"radius\", {\"@type\":\"g:Double\",\"@value\":" + str(self.radius) + "}, \
+                        \"properties\",{\"@type\":\"g:Map\",\"@value\":[\"radius_units\",\"km\"]}]}}}"
+
+        circleJSON = json.loads(circleGSON)
+
+        expectedCircle = self.reader.toObject(circleJSON)
+
+        actualCircle = Circle(self.longitude, self.latitude, self.radius)
+
+        self.assertEqual(expectedCircle, actualCircle)
```

Comment:
Add tests for line and other types (line, polygon, multipoint, multilinestring, multipolygon, geometrycollection)?

Reply:
@sjudeng : After discussion with @FlorianHockmann , I decided that I'll be starting with just Geoshape of Circle and Point. Since other Geoshapes aren't going to be implemented in this release, do we still need to add unit tests for the same?

Reply:
@sjudeng The idea was to keep the scope of the initial PR limited as it's already quite a burden to review a complete project in a single PR, even with only limited functionality.

Reply:
@debasishdebs, @FlorianHockmann Makes sense. I had seen the other types referenced in the documentation but now I see that it was a listing of the possible types and not the supported types. I added a [new issue](https://github.com/JanusGraph/janusgraph-python/issues/9) to track possible future updates to support additional types.